spec/features/admin/unit_price_spec.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+feature '
+    As an admin
+    I want to check the unit price of my products/variants
+' do
+  include AuthenticationHelper
+  include WebHelper
+  
+  let!(:stock_location) { create(:stock_location, backorderable_default: false) }
+  
+  before do
+    allow(OpenFoodNetwork::FeatureToggle).to receive(:enabled?).with(:unit_price, anything) { true }
+  end
+  
+  describe "product", js: true do
+    scenario "creating a new product", :debug do
+      login_as_admin_and_visit spree.admin_products_path
+      click_link 'New Product'
+      select "Weight (kg)", from: 'product_variant_unit_with_scale'
+      fill_in 'product_unit_value_with_description', with: '1'
+      fill_in 'product[price]', with: '1'
+      
+      expect(find("#variant_unit_price", :visible => false).value).to eq '$1.00 / kg'
+    end
+  end
+
+  describe "variant", js: true do
+    scenario "creating a new variant" do
+      product = create(:simple_product, variant_unit: "weight", variant_unit_scale: "1")
+      login_as_admin_and_visit spree.admin_product_variants_path product
+      click_link 'New Variant'
+      fill_in 'unit_value_human', with: '1'
+      fill_in 'variant[price]', with: '1'
+      
+      expect(find("#variant_unit_price", :visible => false).value).to eq '$1,000.00 / kg'
```

Comment:
From reading this test I don't understand why it's "$1,000.00 / kg". Is that because the unit is grams? In that case, it would help a lot if you specify grams in the test setup. In this case it might just be to add an expectation that grams is already selected.

And if we change the default from grams to kg at some point, we would notice that much quicker here. Does that make sense?

Reply:
Actually, this is setup on line 32 with `variant_unit_scale` option
```ruby
product = create(:simple_product, variant_unit: "weight", variant_unit_scale: "1")
```
I agree it's not very straight forward.

Reply:
I added a comment, since I cannot search by label name (because they aren't linked with `for` and `id` attributes) : 
https://github.com/openfoodfoundation/openfoodnetwork/pull/7260/files#diff-c0bd82ce19c4ec544e8a9578fd4c21d2a951a9f2495cae949d86acc7906dc89dR35